src/main/java/ai/chalk/client/GRPCClient.java

```diff
@@ -71,20 +91,150 @@ public GRPCClient(BuilderImpl builder) {
             environmentId = environmentIds.get(0);
         }
 
-        Channel authenticatedServerChannel = Grpc.newChannelBuilder(
-                grpcHost,
-                channelCreds
-        ).maxInboundMessageSize(1024 * 1024 * 100).intercept(
+        Channel authenticatedServerChannel = Grpc.newChannelBuilder(grpcHost, channelCreds)
+            .maxInboundMessageSize(1024 * 1024 * 100)
+            .intercept(
                 new AuthenticatedHeaderClientInterceptor(
                         ServerType.SERVER,
                         Map.of(),
                         tokenRefresher,
                         environmentId
                 )
-        ).build();
-
+            )
+            .build();
         this.teamStub = TeamServiceGrpc.newBlockingStub(authenticatedServerChannel);
 
+        String engineHost;
+        try {
+            engineHost = token
+                .getEnginesOrThrow(environmentId)
+                .replaceFirst("^https?://", "");
+        } catch (Exception e) {
+            throw new ClientException("Error getting engine URI for environment %s".formatted(environmentId), e);
+        }
+        Channel authenticatedEngineChannel = Grpc.newChannelBuilder(engineHost, channelCreds)
+            .maxInboundMessageSize(1024 * 1024 * 500)
+            .intercept(
+                new AuthenticatedHeaderClientInterceptor(
+                        ServerType.ENGINE,
+                        Map.of(),
+                        tokenRefresher,
+                        environmentId
+                )
+            )
+            .build();
+        this.queryStub = QueryServiceGrpc.newBlockingStub(authenticatedEngineChannel);
+    }
+
+    @Override
+    public void printConfig() {
+        System.err.println("Config printing for GRPC client not yet implemented");
```

Comment:
we should get a logger here and use that. when you system.err directly, you subvert how someone might be doing logging, and make it hard for them to get logs into their desired format. i think the best practice for this right now in java is:

```
package util;

import java.lang.System.Logger;
import java.util.Arrays;

public class Testing {
    private static final Logger logger = System.getLogger(Testing.class.getName());

    public static void main(String[] args) {
        logger.log(Logger.Level.INFO, "Started main with arguments: {0}",
                Arrays.toString(args));
    }
}
```

Reply:
addressed